cargo/cargo_build_script_runner/bin.rs

```diff
@@ -53,12 +53,28 @@ fn main() {
 
             let mut command = Command::new(exec_root.join(&progname));
             command
-                .args(args)
                 .current_dir(manifest_dir.clone())
                 .env("OUT_DIR", out_dir_abs)
                 .env("CARGO_MANIFEST_DIR", manifest_dir)
                 .env("RUSTC", rustc);
 
+            while let Some(dep_env_path) = args.next() {
+                if let Ok(contents) = read_to_string(dep_env_path) {
+                    for line in contents.split('\n') {
+                        let mut key_val = line.splitn(2, '=');
+                        match (key_val.next(), key_val.next()) {
+                            (Some(key), Some(value)) => {
+                                command.env(key, value);
+                            }
+                            _ => {
+                                eprintln!("error: Wrong environment file format");
```

Comment:
This is a "should not happen" error, which means if we fall there we have a serious issue in the build rules, maybe leaves some comment to help the user raise this as a bug?

Reply:
Makes sense